controllers/discovery/initDiscoveryUserData.js

```diff
@@ -69,11 +75,21 @@ const InitDiscoveryUserData = async (req, res) => {
     });
     let suggestedUsers = usersWithCommonFollowerResult;
 
+    let totalData = await sequelize.query(totalDataQuery, {
+      type: QueryTypes.SELECT,
+      replacements: {
+        userId
+      },
+      raw: true
+    });
+    totalData = totalData?.[0]?.total_data || 0;
+
     return res.status(200).json({
       success: true,
       message: `Fetch discovery data success`,
       suggestedUsers,
-      nextPage: page + 1
+      nextPage: page + 1,
+      total_page: totalData > 0 && limit > 0 ? Math.ceil(totalData / limit) : 0
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The calculation of `total_page` is correct. However, consider moving this logic to a separate function for better readability and maintainability. Also, the variable name `totalData` could be more descriptive, like `totalUserCount`.
```diff
-    totalData = totalData?.[0]?.total_data || 0;
+    const totalUserCount = totalData?.[0]?.total_data || 0;

-    total_page: totalData > 0 && limit > 0 ? Math.ceil(totalData / limit) : 0
+    total_page: calculateTotalPages(totalUserCount, limit)
```
Where `calculateTotalPages` is a new function:
```javascript
function calculateTotalPages(totalCount, limit) {
  return totalCount > 0 && limit > 0 ? Math.ceil(totalCount / limit) : 0;
}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->